src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js

```diff
@@ -537,22 +537,24 @@ export default class StandardRouteModel extends BaseModel {
      *
      * @for StandardRouteModel
      * @method _updateWaypointsWithPreviousWaypointData
-     * @parma waypointModelList {array<StandardRouteWaypointModel>}
+     * @param waypointModelList {array<StandardRouteWaypointModel>}
      * @private
      */
     _updateWaypointsWithPreviousWaypointData(waypointModelList) {
-        _forEach(waypointModelList, (waypoint, i) => {
-            let previousWaypoint = waypointModelList[i - 1];
-            if (i === 0) {
-                previousWaypoint = waypoint;
+        for (let i = 1; i < waypointModelList.length; i++) {
+            const waypoint = waypointModelList[i];
+            const previousWaypoint = waypointModelList[i - 1];
+
+            let distance = 0;
```

Comment:
this should move up a line and be under `previousWaypoint`.  then there should only be an empty line between `let distance` and the if block below.